sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1769,7 +1769,10 @@ class Dataset[T] private[sql](
    * @since 2.0.0
    */
   def dropDuplicates(colNames: Seq[String]): Dataset[T] = withTypedPlan {
-    val groupCols = colNames.map(resolve)
+    val groupCols = colNames.map {
+      case column if column.contains(".") && !column.contains("`") => s"`$column`"
```

Comment:
Actually we do have a lot of places that trying to resolve column name without parsing it, e.g. `withColumn`, `drop`. We can abstract the logic into a method and call it instead.